solr/solrj/src/java/org/apache/solr/common/util/Utils.java

```diff
@@ -302,10 +302,18 @@ public static byte[] toUTF8(CharArr out) {
   }
 
   public static Object fromJSON(byte[] utf8) {
+    // Need below check in both fromJSON methods since
+    // utf8.length returns a NPE without this check.
+    if (utf8 == null || utf8.length == 0) {
+      return Collections.emptyMap();
+    }
     return fromJSON(utf8, 0, utf8.length);
   }
 
   public static Object fromJSON(byte[] utf8, int offset, int length) {
+    if (utf8 == null || utf8.length == 0 || length == 0) {
```

Comment:
nitpick: the test for `utf8.length == 0` is redundant since it'd be covered by the `length` check.  But whatever.